lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module

```diff
@@ -26,6 +26,26 @@ function dosomething_campaign_form_campaign_node_form_alter(&$form, &$form_state
   dosomething_campaign_run_add_campaign_run_info($form, $form_state);
   // Use #after_build to add JS even on validation errors: https://drupal.org/node/1253520#comment-4881588
   $form['#after_build'][] = 'dosomething_campaign_form_campaign_node_form_after_build';
+
+  $nid = $form['nid']['#value'];
```

Comment:
this might not be set on the `node/add/campaign` page, but this hook still runs. might cause some trouble below